test/DifferentialExpressionMatrixSet_basic_test.py

```diff
@@ -1,173 +1,186 @@
 
 
-def test_get_dem_set(
+@pytest.mark.parametrize(
+    "ref_args",
+    [
+        pytest.param("__SET_REF__", id="set_ref"),
+        pytest.param("__WS_NAME__SET_NAME__", id="ws_name_set_name"),
+    ],
+)
+@pytest.mark.parametrize(
+    "get_method_args",
+    [
+        pytest.param({}, id="empty"),
+        pytest.param({INC_ITEM_INFO: 1}, id="inc_item_info"),
+        pytest.param(
+            {INC_ITEM_REF_PATHS: 1},
+            id="inc_ref_path",
+        ),
+        pytest.param(
+            {
+                INC_ITEM_INFO: 1,
+                INC_ITEM_REF_PATHS: 1,
+            },
+            id="inc_item_info_ref_path",
+        ),
+        pytest.param(
+            {INC_ITEM_INFO: 1, INC_ITEM_REF_PATHS: 1, REF_PATH_TO_SET: ["YES"]},
+            id="inc_item_info_ref_path_w_ref_path",
+        ),
+        pytest.param(
+            {
+                INC_ITEM_INFO: 0,
+                INC_ITEM_REF_PATHS: 0,
+            },
+            id="no_incs",
+        ),
+    ],
+)
+def test_get_diff_exp_matrix_set(
     set_api_client: SetAPI,
     context: dict[str, str | list],
-    ws_id: int,
-    diff_exp_matrix_no_genome_refs: list[str],
+    ws_name: str,
+    ref_args: str,
+    get_method_args: dict[str, str | int],
+    differential_expression_matrix_with_genome_set,
+    differential_expression_matrix_no_genome_set,
 ) -> None:
-    set_name = "test_expression_set"
-    set_items = [{"label": "wt", "ref": ref} for ref in diff_exp_matrix_no_genome_refs]
-    dem_set = {"description": "test_test_diffExprMatrixSet", "items": set_items}
-    dem_set_ref = set_api_client.save_differential_expression_matrix_set_v1(
-        context,
-        {
-            "workspace_id": ws_id,
-            "output_object_name": set_name,
-            "data": dem_set,
-        },
-    )[0]["set_ref"]
-
-    fetched_set = set_api_client.get_differential_expression_matrix_set_v1(
-        context, {"ref": dem_set_ref, INC_ITEM_INFO: 0}
-    )[0]
-    assert fetched_set is not None
-    assert "data" in fetched_set
-    assert "info" in fetched_set
-    assert len(fetched_set["data"]["items"]) == len(diff_exp_matrix_no_genome_refs)
-    assert dem_set_ref == info_to_ref(fetched_set["info"])
-    for item in fetched_set["data"]["items"]:
-        assert "info" not in item
-        assert "ref" in item
-        assert "ref_path" not in item
-        assert "label" in item
-
-    fetched_set_with_info = set_api_client.get_differential_expression_matrix_set_v1(
-        context, {"ref": dem_set_ref, INC_ITEM_INFO: 1}
-    )[0]
-    assert fetched_set_with_info is not None
-    assert "data" in fetched_set_with_info
-    for item in fetched_set_with_info["data"]["items"]:
-        assert "info" in item
-        assert "ref" in item
-        assert "label" in item
+    for saved_set in [
+        differential_expression_matrix_with_genome_set,
+        differential_expression_matrix_no_genome_set,
+    ]:
+        check_get_set(
```

Comment:
Ditto for getting sets -- runs all the checks for both the DEM with genome and DEM without.